UKHomeOffice/nmsw-ui
# NMSW-192 Clear values and errors from conditional fields on change to radio

# Ticket

NMSW-192

----

## AC

Given I have entered a value in a conditional field
When I refresh the page
And click on a different radio option
Then the original conditional field should be hidden
And the value stored in the session for that field should be null
And the value stored in the formData for that field should be null

Given I have clicked on a radio with a conditional field
And have not entered a value
And the field is required when visible
When I click submit
Then an error should show for that conditional field

Given the error is showing for that conditional field
When I click on another radio option
Then the error for the conditional field should be cleared

----

## To test
1.

- click on dog
- enter data into conditional
- refresh page
- click on rabbit (or cat or other)
- click on submit
- > check formData (you may need to add a console log to check this for now), the value for `dogBreed` should be null
- > check the session storage, the value for `dogBreed` should be null

2. 

- click on cat
- submit form
-> error for cat breed should show in summary and field
- click dog
- > error for cat breed should not show in summary
- click cat again
- > error for cat breed should not show on conditional

----

## Notes
